core/src/test/java/org/apache/spark/util/collection/TestTimSort.java

```diff
@@ -0,0 +1,111 @@
+package org.apache.spark.util.collection;
+
+import java.util.*;
+
+/**
+ * This codes generates a int array which fails the standard TimSort, Borrowed from
+ * the reporter of this bug.
+ *
+ * http://www.envisage-project.eu/timsort-specification-and-verification/
```

Comment:
Yes that's good practice. For completeness, because this AL2 and because there is no additional NOTICE file, we don't need to do anything else to our LICENSE or NOTICE file.